infertrade/utilities/operations.py

```diff
@@ -413,21 +428,19 @@ def restricted_function(*args, **kwargs) -> pd.DataFrame:
         # Get allocation dataframe from allocation function
         dataframe = allocation_function(*args, **kwargs)
         
-        
-        # Restrictions by limiting allocations between range
-
-        allocation_lower_limit=0
-        allocation_upper_limit=1
-
-        if "allocation_lower_limit" in kwargs:
+        # Restriction by limiting allocations between range
+        if "allocation_lower_limit" in kwargs and "allocation_upper_limit" in kwargs:
```

Comment:
This will only apply the restriction if both restrictions keywords are present. We could apply them independently one by one.